src/components/opt-in-out/helpers/make-beta-welcome-toast.tsx

```diff
@@ -0,0 +1,43 @@
+import { IconArrowUpRight16 } from '@hashicorp/flight-icons/svg-react/arrow-up-right-16'
+import { toast, ToastColor } from 'components/toast'
+import Button from 'components/button'
+import { ToastDisplayProps } from 'components/toast/components/toast-display/types'
+import { OptInPlatformOption, PlatformOptionTitles } from '../types'
+
+export default function makeBetaWelcomeToast(
+  optInPlatform: OptInPlatformOption,
+  handleOptOut: () => void
+) {
+  // Get the product name
+  const platformName = PlatformOptionTitles[optInPlatform]
+  // Make toast
+  toast({
+    color: ToastColor.highlight,
+    title: `Welcome to the ${platformName} Website Beta!`,
+    description:
+      'You can leave the Beta at anytime by clicking the Leave Beta button near the start of the main content area.',
+    renderActions: ({
+      dismissSelf,
+    }: {
+      dismissSelf: ToastDisplayProps['dismissSelf']
+    }) => (
+      <>
+        <Button
+          color="secondary"
+          onClick={dismissSelf}
+          size="small"
+          text="Continue"
+        />
+        <Button
+          color="tertiary"
+          icon={<IconArrowUpRight16 />}
+          iconPosition="trailing"
+          onClick={handleOptOut}
+          size="small"
+          text="Leave beta"
+        />
+      </>
+    ),
```

Comment:
🙋🏻 What led you to the approach of making this a function rather than array of `ReactNode`s or a single `ReactNode` via a `Fragment`?

Reply:
Great question! 💖 The intent here was to make the `dismissSelf` function, which is declared within the context of `components/toast`, available to the actions.

I'd had a single `ReactNode` as the prop type previously, but that seemed to limit the ability to make `dismissSelf` available to the `actions`.

I hadn't considered the array of `ReactNode` too much, it's not a pattern I've seen often! Seems neat though, might align with the layout intent of the slot here (which kinda does want an array of actions to render in a flex layout; rather than having to trust the consumer to set that up correctly).

Reply:
This makes sense! Thanks for the extra context. 😊